Sources/Controllers/Cells/OAFolderCardsCell.m

```diff
@@ -21,7 +21,8 @@ @interface OAFolderCardsCell() <UICollectionViewDelegate, UICollectionViewDataSo
 @implementation OAFolderCardsCell
 {
     NSMutableArray *_data;
-    int _selectedItemIndex;
+    NSInteger _selectedItemIndex;
```

Comment:
Seems like you do not need two variables here. ScrollToItem method should simply check against the current selected index. The cache is redundant